content_script.js

```diff
@@ -1,93 +1,111 @@
+    } else if (tagName && this.FILTERED_TAGS.has(tagName.toLowerCase())) {
+      // filter out elements that are in filteredTags.
     } else if (element.nodeType == Node.TEXT_NODE) {
       this.html.push(element.textContent);
     } else {
       this.html.push(new Array(depth+1).join('  '));
-      this.html.push(`<${element.tagName.toLowerCase()} `);
+      this.html.push(`<${tagName.toLowerCase()} `);
 
-      var style = element.ownerDocument.defaultView.getComputedStyle(element, null).cssText;
+      var win = element.ownerDocument.defaultView;
+      var style = win.getComputedStyle(element, null).cssText;
       this.styleIndices.push(this.html.length);
       this.html.push(`style="${style}" `);
 
       var attributes = element.attributes;
       if (attributes) {
         for (var i = 0, attribute; attribute = attributes[i]; i++) {
-          if (attribute.name.toLowerCase() == 'src') {
-            if (element.tagName.toLowerCase() != 'iframe') {
-              this.html.push(`${attribute.name}=`);
-              this.html.push(''); // entry where properly escaped quotes will go
-              this.srcHoles[this.html.length] = attribute.value; // mark location to insert url later
-              this.html.push(''); // entry where data url will go
-              this.html.push(''); // entry where properly escaped quotes will go
-              this.html.push(' ');
-            }
-          } else if (attribute.name.toLowerCase() != 'style') {
-            this.html.push(`${attribute.name}="${attribute.value}" `);
+          switch (attribute.name.toLowerCase())  {
+            case 'src':
+              if (tagName.toLowerCase() != 'iframe') {
+                this.html.push(`${attribute.name}=`);
+                this.srcHoles[this.html.length] = attribute.value;
+                this.html.push(''); // entry where data url will go.
+                this.html.push(' ');
+              }
+            case 'style':
+              break;
+            default:
+              this.html.push(`${attribute.name}="${attribute.value}" `);
           }
         }
-        if (element.tagName.toLowerCase() == 'iframe') { // TODO: IS IT POSSIBLE AN IFRAME CAN HAVE NO ATTRIBUTES?
+        // TODO(sfine): ensure this is working by making sure that an iframe
+        //              will always have attributes.
+        if (tagName.toLowerCase() == 'iframe') {
           this.html.push('srcdoc=');
-          this.frameHoles[this.html.length] = this.iframeFullPosition(window) + '.' + this.iframePosition(element.contentWindow);
-          this.html.push(''); // entry where the iframe contents will go
+          var path = this.iframeFullyQualifiedName(window);
+          var index = this.iframeIndex(element.contentWindow);
+          this.frameHoles[this.html.length] = path + '.' + index;
+          this.html.push(''); // entry where the iframe contents will go.
           this.html.push(' ');
```

Comment:
I decided to simply delete it.  This will always be the last attribute, so there is no need to add a space before the closing bracket.